checks.d/activemq_xml.py

```diff
@@ -42,27 +42,38 @@ def check(self, instance):
         detailed_queues = instance.get("detailed_queues", [])
         detailed_topics = instance.get("detailed_topics", [])
         detailed_subscribers = instance.get("detailed_subscribers", [])
+        suppress_errors = bool(instance.get("suppress_errors", None))
 
         tags = custom_tags + ["url:{0}".format(url)]
 
         self.log.debug("Processing ActiveMQ data for %s" % url)
-        data = self._fetch_data(url, QUEUE_URL, username, password)
-        self._process_data(data, "queue", tags, max_queues, detailed_queues)
+        data = self._fetch_data(url, QUEUE_URL, username, password, suppress_errors)
+        if(data):
```

Comment:
why not just do:
```python
if not data and suppress_errors:
    # We don't have any data and we are suppressing errors. Let's just stop the check here
    return
```

This way you don't have to check for the data that many  times.
What do you think ?